core/src/commonMain/kotlin/org/aquila3d/core/scene/Node.kt

```diff
@@ -0,0 +1,40 @@
+package org.aquila3d.core.scene
+
+import org.aquila3d.core.math.Matrix4
+import org.aquila3d.core.math.MutableMatrix4
+import org.aquila3d.core.math.Quaternion
+import org.aquila3d.core.math.Vector3
+
+class Node(val parent: Node? = null) {
+
+    /**
+     * The translation of this [Node] relative to its parent (if it has one)
+     */
+    private val position: Vector3 = Vector3()
+
+    /**
+     * The scale of this [Node] relative to its parent. Defaults to Unity.
+     */
+    private val scale: Vector3 = Vector3.ONES()
+
+    /**
+     * The orientation of this [Node] about its centroid.
+     */
+    private val orientation: Quaternion = Quaternion()
+
+    /**
+     * [Set] of child [Node]s.
+     */
+    private val children: MutableSet<Node> = mutableSetOf()
+
+    private val scratchModelMatrix: MutableMatrix4 = MutableMatrix4()
```

Comment:
This seems like a potential source of bugs keeping this around. Are you certain it's necessary to hold this?

Reply:
My intention with holding it was to prevent GC thrashing as every Node in the scene will require a matrix update per frame in the worst case scenario and each renderable object in the scene has an associated node in a 1:1 relationship